sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -407,8 +407,8 @@ private[sql] object QueryParsingErrors extends QueryErrorsBase {
 
   def computeStatisticsNotExpectedError(ctx: IdentifierContext): Throwable = {
     new ParseException(
-      errorClass = "_LEGACY_ERROR_TEMP_0036",
-      messageParameters = Map("ctx" -> ctx.getText),
+      errorClass = "INVALID_SQL_SYNTAX",
+      messageParameters = Map("inputString" -> s"${ctx.getText} must be ${toSQLStmt("NOSCAN")}"),
```

Comment:
It would be better to do not embed error's text in source code. Could you leave the error class as a separate one (+ assign name) or at least make it as a sub-class of `INVALID_SQL_SYNTAX`.

Reply:
I have found many similar uses in `QueryParsingErrors`. Should we refactor it to classify `INVALID_SQL_SYNTAX ` and avoid to embed error's text in source code?
eg: https://github.com/apache/spark/blob/df7e2d07f04c7e16af04c1292227a90516083de1/sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala#L74-L80

Because I think it is not good to assign a new name for the above scenario, because it is a syntax error.

Reply:
> Should we refactor it to classify INVALID_SQL_SYNTAX  and avoid to embed error's text in source code?

Yep. Let's refactor `INVALID_SQL_SYNTAX` first of all, and then come back to this PR.

Reply:
👌🏻

Reply:
@panbingkun Could you continue the work on the PR.

Reply:
Yes, I will continue the work on the PR right now.